source/extensions/stat_sinks/graphite_statsd/config.cc

```diff
@@ -38,9 +38,9 @@ GraphiteStatsdSinkFactory::createStatsSink(const Protobuf::Message& config,
   }
   case envoy::extensions::stat_sinks::graphite_statsd::v3::GraphiteStatsdSink::StatsdSpecifierCase::
       STATSD_SPECIFIER_NOT_SET:
-    break; // Fall through to PANIC.
+    break;
   }
-  PANIC_DUE_TO_CORRUPT_ENUM;
+  throw EnvoyException("unexpected statsd specifier enum");
```

Comment:
probably stupid question but I don't know this code, are we sure this is called from the control plane and not the data plane? In general throwing an exception from the data plane may cause a crash anyway and I'm not sure what our policy is around adding that.

This applies to the throw statements below as well.

Reply:
I assumed any of these factory classes that take in a proto object to construct something should be fine for throwing exceptions.

Here's the only place I see this being being called: https://github.com/envoyproxy/envoy/blob/main/source/server/configuration_impl.cc#L121

Let me double check on the other instances.

Reply:
Yes, it is ok to throw here. The documentation for the `createStatsSink` interface says so too.